strawberry/field.py

```diff
@@ -47,6 +47,18 @@
 UNRESOLVED = object()
 
 
+def _is_generic(resolver_type: Union[StrawberryType, type]) -> bool:
+    """Returns True if `resolver_type` is generic else False"""
+    if isinstance(resolver_type, StrawberryType):
+        return resolver_type.is_generic
+
+    # solves the Generic subclass case
+    if hasattr(resolver_type, "_type_definition"):
+        return resolver_type._type_definition.is_generic  # type: ignore
+
+    return False
```

Comment:
adding a note for the future (maybe), we could move this method on the resolver itself potentially